src/docs/tutorial-transaction-process/customize-pricing/index.md

```diff
@@ -618,153 +620,88 @@ the booking.
+function called _getOrderParams_, which creates the correct set of order
+parameters for all line item related API calls. Let's add cleaning fee
+handling to that function.
 
 ```shell
 └── src
     └── containers
         └── CheckoutPage
-            └── CheckoutPage.js
+            └── CheckoutPageWithPayment.js
 ```
 
 ```diff
-loadInitialData() {
-  ...
-      const deliveryMethod = pageData.orderData?.deliveryMethod;
-+     const hasCleaningFee = pageData.orderData?.cleaningFee?.length > 0;
-      fetchSpeculatedTransaction(
-        {
-          listingId,
-          deliveryMethod,
-+         hasCleaningFee,
-          ...quantityMaybe,
-          ...bookingDatesMaybe(pageData.orderData.bookingDates),
-        },
-        processAlias,
-        transactionId,
-        requestTransition,
-        isPrivileged
-      );
-...
-```
-
-This function call dispatches a `speculateTransaction` action in
-`CheckoutPage.duck.js`, which in turn calls the template server using
-the correct endpoint.
+const getOrderParams = (pageData, shippingDetails, optionalPaymentParams, config) => {
+  const quantity = pageData.orderData?.quantity;
+  const quantityMaybe = quantity ? { quantity } : {};
+  const deliveryMethod = pageData.orderData?.deliveryMethod;
+  const deliveryMethodMaybe = deliveryMethod ? { deliveryMethod } : {};
++ const hasCleaningFee = pageData.orderData?.cleaningFee?.length > 0;
+
+  const { listingType, unitType } = pageData?.listing?.attributes?.publicData || {};
+  const protectedDataMaybe = {
+    protectedData: {
+      ...getTransactionTypeData(listingType, unitType, config),
+      ...deliveryMethodMaybe,
+      ...shippingDetails,
+    },
+  };
 
-```shell
-└── src
-    └── containers
-        └── CheckoutPage
-            └── CheckoutPage.duck.js
+  // These are the order parameters for the first payment-related transition
+  // which is either initiate-transition or initiate-transition-after-enquiry
+  const orderParams = {
+    listingId: pageData?.listing?.id,
+    ...deliveryMethodMaybe,
++   hasCleaningFee,
+    ...quantityMaybe,
+    ...bookingDatesMaybe(pageData.orderData?.bookingDates),
+    ...protectedDataMaybe,
+    ...optionalPaymentParams,
+  };
+  return orderParams;
+};
 ```
 
-To pass the cleaning fee selection to the API call, we add it to
-`orderData` within the `speculateTransaction` action.
-
-```diff
-export const speculateTransaction = (
-  ...
-- const { deliveryMethod, quantity, bookingDates, ...otherOrderParams } = orderParams;
-+ const { deliveryMethod, quantity, bookingDates, hasCleaningFee, ...otherOrderParams } = orderParams;
-...
-
-  // Parameters only for client app's server
-- const orderData = deliveryMethod ? { deliveryMethod } : {};
-+ const orderData = deliveryMethod || hasCleaningFee ? { deliveryMethod, hasCleaningFee } : {};
-
-```
+This function is used to build order parameters when the component loads
+initial data for the page, and the order params are then passed to a
```

Comment:
Comment:
> This function is used to build order **parameters** when the component loads
> initial data for the page, and the order **params** are then passed to a

This is a small detail but should we stick to using either `params` or `parameters`?

Reply:
Good point, unified!